### PR TITLE
fix: OpenTelemetry bugfixes

### PR DESCRIPTION
**Pull request by evansims**
## Description

This PR fixes several bugs identified in the recent OpenTelemetry improvements:

- Fixed `http_client_request_duration` being reported in seconds rather than the intended milliseconds
- Fixed the sync client mistakenly passing the entire configuration (rather than just the OpenTelemetry configuration as intended) to `queryDuration()` and `requestDuration()`.
- Fixed an issue in which some attributes may not have been exported as expected under some conditions
- All attribute values are now correctly exported as their intended types rather than defaulting to strings
- Fixed an issue in which `queryDuration()` and `requestDuration()` may not have updated their histograms reliably when `attr_http_client_request_duration` or `attr_http_server_request_duration` (respectively) were not enabled (which is the default)

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by ewanharris — APPROVED** [review]

File: openfga_sdk/telemetry/metrics.py [hunk 1]
```diff
@@ -122,12 +122,27 @@ def requestDuration(
...
-            isinstance(configuration, TelemetryMetricsConfiguration) is False
+            isinstance(configuration, TelemetryConfiguration) is False
             or configuration.metrics.histogram_request_duration.enabled is False
             or configuration.metrics.histogram_request_duration.attributes() == {}
         ):
             return self.histogram(TelemetryHistograms.request_duration)
 
+        if (
+            value is None
+            and TelemetryAttributes.http_client_request_duration.name in attributes
```
Comment (rhamzeh) [inline openfga_sdk/telemetry/metrics.py:133]: Can we drop the server and client request durations as attributes?

**Review by rhamzeh — CHANGES_REQUESTED** [review]
Can you also please:
* Update the documentation page with
	1. the defaults
	2. How to configure it
* Update the example with a sample configuration?
* Add validation that they're not configuring with invalid meter/attribute names?

File: openfga_sdk/telemetry/attributes.py [hunk 2]
```diff
@@ -69,21 +85,44 @@ def prepare(
...
         if attributes is not None:
             for attribute, value in attributes.items():
-                if isinstance(attribute, TelemetryAttribute):
-                    if filter is not None and attribute not in filter:
-                        continue
+                if value is None:
+                    continue
+
+                if isinstance(attribute, str):
+                    attribute = attribute.lower().replace("-", "_").replace(".", "_")
```
Comment (rhamzeh) [inline openfga_sdk/telemetry/attributes.py:92]: What is this for?

Comment (evansims) [reply openfga_sdk/telemetry/attributes.py:92]:
```python
def prepare(
    self,
    attributes: dict[TelemetryAttribute | str, str | int] | None,
    filter: list[TelemetryAttribute | str] | None = None,
) -> dict[str, str | int]:
```

`attributes` can either be a TelemetryAttribute instance or a string. This allows us to provide `fga-client.request.client_id` as the attribute key and properly resolve it to an instance of the `fga_client_request_client_id` TelemetryAttribute.

File: openfga_sdk/telemetry/metrics.py (hunk 1 above)
Comment (evansims) [reply openfga_sdk/telemetry/metrics.py:133]: Can you clarify? They're opt-in attributes presently; I'd imagine we'd want to keep them for alternate histograms, counters, etc?

**Comment by evansims** [comment]
> Can you also please:
> 
> * Update the documentation page with
>   
>   1. the defaults
>   2. How to configure it
> * Update the example with a sample configuration?
> * Add validation that they're not configuring with invalid meter/attribute names?

I'd prefer to do a documentation pass as a separate PR, but yes, that's on my radar

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**